src/nix/command.cc

```diff
@@ -131,11 +131,18 @@ void MixProfile::updateProfile(const Buildables & buildables)
     std::optional<StorePath> result;
 
     for (auto & buildable : buildables) {
-        for (auto & output : buildable.outputs) {
-            if (result)
-                throw Error("'--profile' requires that the arguments produce a single store path, but there are multiple");
-            result = output.second;
-        }
+        std::visit(overloaded {
+            [&](BuildableOpaque bo) {
+                result = bo.path;
```

Comment:
Need to check that there is only one `result` here.

Reply:
Fixed this, and with less duplication, by accumulating everything in a vector, and the throwing an error if the vector's length is not exactly 1.